kie-maven-plugin/src/main/java/org/kie/maven/plugin/GenerateANCMojo.java

```diff
@@ -0,0 +1,192 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.maven.plugin;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.DependencyResolutionRequiredException;
+import org.apache.maven.artifact.resolver.filter.CumulativeScopeArtifactFilter;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+import org.drools.ancompiler.CompiledNetworkSource;
+import org.drools.ancompiler.ObjectTypeNodeCompiler;
+import org.drools.compiler.kproject.ReleaseIdImpl;
+import org.drools.core.impl.InternalKnowledgeBase;
+import org.drools.modelcompiler.CanonicalKieModule;
+import org.kie.api.KieServices;
+import org.kie.api.runtime.KieContainer;
+
+import static org.kie.maven.plugin.ExecModelMode.ancEnabled;
+import static org.kie.maven.plugin.ExecModelMode.isModelCompilerInClassPath;
+
+@Mojo(name = "generateANC",
+        requiresDependencyResolution = ResolutionScope.NONE,
+        defaultPhase = LifecyclePhase.COMPILE)
+public class GenerateANCMojo extends AbstractDMNValidationAwareMojo {
+
+    @Parameter(defaultValue = "${session}", required = true, readonly = true)
+    private MavenSession mavenSession;
+
+    @Parameter(required = true, defaultValue = "${project.build.directory}")
+    private File targetDirectory;
+
+    @Parameter(required = true, defaultValue = "${project.basedir}")
+    private File projectDir;
+
+    @Parameter(required = true, defaultValue = "${project.build.testSourceDirectory}")
+    private File testDir;
+
+    @Parameter
+    private Map<String, String> properties;
+
+    @Parameter(required = true, defaultValue = "${project}")
+    private MavenProject project;
+
+    @Parameter(required = true, defaultValue = "${project.build.outputDirectory}")
+    private File outputDirectory;
+
+    @Parameter(property = "generateModel", defaultValue = "YES_WITHDRL")
+    private String generateModel;
```

Comment:
you might want to "move up" similarly to here: https://github.com/kiegroup/droolsjbpm-integration/blob/32f44c724770dfd3eca136d3cacdcf0b90455a1e/kie-maven-plugin/src/main/java/org/kie/maven/plugin/AbstractDMNValidationAwareMojo.java#L63-L68